phoenix-core-client/src/main/java/org/apache/phoenix/util/json/BsonJsonProvider.java

```diff
@@ -0,0 +1,250 @@
+    @Override
+    public Object parse(InputStream jsonStream, String charset) throws InvalidJsonException {
+        return null;
+    }
+
+    @Override
+    public String toJson(Object obj) {
+        return null;
+    }
+
+    @Override
+    public Object createArray() {
+        return new BsonArray();
+    }
+
+    @Override
+    public boolean isArray(final Object obj) {
+
+        return (obj instanceof BsonArray || obj instanceof List);
+    }
+
+    @Override
+    public Object getArrayIndex(final Object obj, final int idx) {
+
+        return toBsonArray(obj).get(idx);
+    }
+
+    @Override
+    public void setArrayIndex(final Object array, final int index, final Object newValue) {
+        if (!isArray(array)) {
+            throw new UnsupportedOperationException();
+        } else {
+            BsonArray arr = toBsonArray(array);
+            if (index == arr.size()) {
+                arr.add(toBsonValue(newValue));
+            } else {
+                arr.set(index, toBsonValue(newValue));
+            }
+        }
+    }
+
+    @Override
+    public Object createMap() {
+        return new BsonDocument();
+    }
+
+    @Override
+    public boolean isMap(final Object obj) {
+        return (obj instanceof BsonDocument);
+    }
+
+    @Override
+    public Object getMapValue(final Object obj, final String key) {
+        BsonDocument bsonDocument = toBsonDocument(obj);
+        Object o = bsonDocument.get(key);
+        if (!bsonDocument.containsKey(key)) {
+            return UNDEFINED;
+        } else {
+            return unwrap(o);
+        }
+    }
+
+    @Override
+    public Iterable<?> toIterable(final Object obj) {
+        BsonArray arr = toBsonArray(obj);
+        List<Object> values = new ArrayList<Object>(arr.size());
+        for (Object o : arr) {
+            values.add(toJavaType(toBsonValue(o)));
+        }
+        return values;
+    }
+
+    @Override
+    public void setProperty(final Object obj, final Object key, final Object value) {
+        if (isMap(obj)) {
+            toBsonDocument(obj).put(key.toString(), toBsonValue(value));
+        } else {
+            BsonArray array = toBsonArray(obj);
+            int index;
+            if (key != null) {
+                index = key instanceof Integer ? (Integer) key : Integer.parseInt(key.toString());
+            } else {
+                index = array.size();
+            }
+
+            if (index == array.size()) {
+                array.add(toBsonValue(value));
+            } else {
+                array.set(index, toBsonValue(value));
+            }
+        }
+    }
+
+    private static BsonArray toBsonArray(final Object o) {
+        return (BsonArray) o;
+    }
+
+    private static BsonDocument toBsonDocument(final Object o) {
+        return (BsonDocument) o;
+    }
+
+    /**
+     * https://github.com/spring-projects/spring-data-mongodb/blob/main/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/util/BsonUtils.java#L66
```

Comment:
What is the purpose of this link here (line 66?)?

Reply:
This is the reference code for this function implementation. We can actually remove this.